utils/tracefs.c

```diff
@@ -1,67 +1,50 @@
 #include <fcntl.h>
+#include <linux/magic.h>
+#include <mntent.h>
 #include <stdio.h>
 #include <string.h>
+#include <sys/vfs.h>
 
 #include "utils/tracefs.h"
 #include "utils/utils.h"
 
-#define PROC_MOUNTINFO "/proc/self/mountinfo"
+#define PROC_MOUNTS_DIR_PATH "/proc/mounts"
+#define TRACEFS_DIR_PATH "/sys/kernel/tracing"
 
 static char *TRACING_DIR = NULL;
 
 static bool find_tracing_dir(void)
 {
 	FILE *fp;
-	char *line = NULL, fs_type[NAME_MAX], mount_point[PATH_MAX];
-	static char debugfs_suffix[] = "tracing";
-	bool debugfs_found = false;
-	size_t len;
+	struct mntent *ent;
+	struct statfs fs;
 
 	if (TRACING_DIR)
-		return false;
+		return true;
+
+	if (!statfs(TRACEFS_DIR_PATH, &fs) && fs.f_type == TRACEFS_MAGIC) {
+		xasprintf(&TRACING_DIR, "%s", TRACEFS_DIR_PATH);
+		return true;
+	}
```

Comment:
Could you add one more quick check logic for `OLD_TRACEFS_DIR_PATH "/sys/kernel/debug/tracing"`?

Reply:
> Also it's not true that having debugfs at /sys/kernel/debug guarantees tracefs at /sys/kernel/debug/tracing. Actually you don't need to check /sys/kernel/debug, just check if /sys/kernel/debug/tracing is tracefs type.

Incorporated feedback based on that comment.
Added code to check if the path `/sys/kernel/debug/tracing` is of type `tracefs`.

Additionally, I've also added a link to the official kernel repository with a 12-character representation of the length of the commit hash.